### ISSUE TITLE
Implement a `test` verb

### ISSUE THREAD

**Issue by drewoldag** [lincc-frameworks/fibad#628]
**Feature request**
`hyrax.test()` would be run after `hyrax.train()` perhaps as a final check after hyperparameter tuning. 

The user should be able to specify via a config toml file the directory to use that contains the model weights as well as the testing split index file. If not specified, it is assumed that test would use the most recent output of training. 

It is assumed that the user would have generated a train/validate/test split when running `h.train()` at least on the last during the training run that produced the model weights to be tested. 

The test verb should load the model weight in eval mode, and would utilize the train_step method in the model in order to calculate the loss (or user defined) metric. That metric, or metrics, should be logged to tensorboard, and probably logged to the command line too.
